### PR TITLE
Minor changes to CMake comments, etc.

### PR DESCRIPTION
**Pull request by derobins**
Minor tidying of things noticed while documenting CMake files

### CODE REVIEW

File: config/cmake/mccacheinit.cmake [hunk 1]
```diff
@@ -11,9 +11,9 @@
 #
 # This is the CMakeCache file.
 
-########################
+#########################
 # EXTERNAL cache entries
```
Comment (byrnHDF) [inline config/cmake/mccacheinit.cmake:15]: mc stands for "memory check".

**Comment by byrnHDF** [comment]
One minor note on usage of case in naming. There are some packaging conventions that use lower case intentionally.
(Nothing in this PR)

File: config/cmake/mccacheinit.cmake (hunk 1 above)
Comment (lrknox) [reply config/cmake/mccacheinit.cmake:15]: So line 15 should be "# EXTERNAL cache entries for memory checking"?

Comment (byrnHDF) [reply config/cmake/mccacheinit.cmake:15]: Or something like it.

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**